LoopKitUI/OnboardingUI.swift

```diff
@@ -0,0 +1,87 @@
+    ///
+    /// - Parameters:
+    ///     - identifier: The identifier of the pump manager to setup.
+    /// - Returns: Either a conforming view controller to create and setup the pump manager, a newly created and setup pump manager, or an error.
+    func setupPumpManager(withIdentifier identifier: String, initialSettings settings: PumpManagerSettings) -> UIResult<UIViewController & PumpManagerCreateNotifying & PumpManagerOnboardNotifying & CompletionNotifying, PumpManager, Error>
+}
+
+public protocol ServiceProvider: AnyObject {
+    /// The active services.
+    var activeServices: [Service] { get }
+
+    /// This descriptor list of available services.
+    var availableServices: [ServiceDescriptor] { get }
+
+    /// Setup the setup with the specified identifier.
+    ///
+    /// - Parameters:
+    ///     - identifier: The identifier of the setup to setup.
+    /// - Returns: Either a conforming view controller to create and setup the service, a newly created and setup service, or an error.
+    func setupService(withIdentifier identifier: String) -> UIResult<UIViewController & ServiceCreateNotifying & ServiceOnboardNotifying & CompletionNotifying, Service, Error>
+}
+
+public protocol OnboardingDelegate: AnyObject {
+    /// Informs the delegate that onboarding has new therapy settings that should be persisted.
+    ///
+    /// - Parameters:
+    ///     - notifying: The source of the notification.
+    ///     - therapySettings: The new therapy settings.
+    func onboardingNotifying(_ notifying: OnboardingNotifying, hasNewTherapySettings therapySettings: TherapySettings)
+}
+
+public protocol OnboardingNotifying: AnyObject {
+    /// Delegate to notify about onboarding changes.
+    var onboardingDelegate: OnboardingDelegate? { get set }
+}
+
+public protocol OnboardingUI: Onboarding {
+    /// Provides a view controller to configure onboarding, if needed.
+    ///
+    /// - Parameters:
+    ///   - preferredGlucoseUnit: The preferred glucose unit.
+    ///   - cgmManagerProvider: The provider of CGM Manager functionality.
+    ///   - pumpManagerProvider: The provider of Pump Manager functionality.
+    ///   - serviceProvider: The provider of Service functionality.
+    ///   - colorPalette: The colors to use in any UI,
+    /// - Returns: A view controller to create and configure a new onboarding.
+    func onboardingViewController(preferredGlucoseUnit: HKUnit,
+                                  cgmManagerProvider: CGMManagerProvider,
+                                  pumpManagerProvider: PumpManagerProvider,
+                                  serviceProvider: ServiceProvider,
+                                  colorPalette: LoopUIColorPalette) -> (UIViewController & OnboardingNotifying & CGMManagerCreateNotifying & CGMManagerOnboardNotifying & PumpManagerCreateNotifying & PumpManagerOnboardNotifying & ServiceCreateNotifying & ServiceOnboardNotifying & CompletionNotifying)
```

Comment:
Does the returned VC need to observe changes to the `preferredGlucoseUnit`?

Reply:
Yes, I suppose it does. By extension, though, shouldn't the `CGMManagerUI.setupViewController` also observe changes?

Reply:
If preferred glucose unit is being passed into the `CGMManagerUI.setupViewController`, then yes.

Reply:
It isn't right now, but shouldn't it be? If the CGM manager **settings** UI is going to update based upon changing preferred glucose units, should the **setup** UI also update accordingly?

Reply:
Yes, if the unit is needed. Currently, there is no use case with either the mock CGM or Dexcom during setup. However, for both of their settings, they display glucose values.

I'm not against adding it, I'm just pointing out there isn't any use yet.